### ISSUE TITLE
Logging should be done with same log level used in condition and log statement

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#501]
We should report a warning if a log method is invoked inside an if block but does have a different log level than the one checked inside the if block.

E.g.:
Following should report a warning:
```c#
if (Log.IsInfoEnabled)
{
    Log.Debug("something");
}
```

In addition, following should report a warning:
```c#
if (Log.IsDebugEnabled)
{
    Log.Info("something");
}
```
